File: prod/frontend/src/app/shared/interfaces/observation.ts
export interface Observation {
    country: string;
    epoch: number;
    humidity: number;
    lat: number;
    lon: number;
    metric: {
        temp: number;
        heatIndex: number;
        dewpt: number;
        windChill: number;
        windSpeed: number;
        pressure:number
        precipTotal:number,
        // Add more properties as needed
    };
    neighborhood: string;
    obsTimeLocal: string;
    obsTimeUtc: string;
    qcStatus: number;
    realtimeFrequency: null | any; // Modify the type as needed
    softwareType: string;
    solarRadiation: number;
    stationID: string;
    uv: number;
    winddir: number;
    // Add more properties as needed
}

export interface ObservationData {
    observations: Observation[];
}

File: prod/WeatherServiceSolution/WeatherService/Models/Observation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WeatherService.Models
{
    public class WeatherDataWeatherUnderground
    {
        public Observation[] observations { get; set; }
    }

    public class Observation
    {
        public string stationID { get; set; }
        public string obsTimeUtc { get; set; }
        public string obsTimeLocal { get; set; }
        public string neighborhood { get; set; }
        public string softwareType { get; set; }
        public string country { get; set; }
        public float? solarRadiation { get; set; }
        public float lon { get; set; }
        public object realtimeFrequency { get; set; }
        public int epoch { get; set; }
        public float lat { get; set; }
        public int? uv { get; set; }
        public int winddir { get; set; }
        public int humidity { get; set; }
        public int qcStatus { get; set; }
        public Metric metric { get; set; }
    }

    public class Metric
    {
        public int temp { get; set; }
        public int heatIndex { get; set; }
        public int dewpt { get; set; }
        public int windChill { get; set; }
        public int windSpeed { get; set; }
        public int windGust { get; set; }
        public float pressure { get; set; }
        public object precipRate { get; set; }
        public float precipTotal { get; set; }
        public int elev { get; set; }
    }
}
